smartlead/delete.go

```diff
@@ -0,0 +1,41 @@
+package smartlead
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+// Delete removes object. As of now only removal of Campaigns is allowed.
+func (c *Connector) Delete(ctx context.Context, config common.DeleteParams) (*common.DeleteResult, error) {
+	if len(config.ObjectName) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	if len(config.RecordId) == 0 {
+		return nil, common.ErrMissingRecordID
+	}
+
+	if config.ObjectName != objectNameCampaign {
```

Comment:
Let's add this into a list and put the list above the function so that it's easier to add in new objects when they do support it